src/components/dao/dao.js

```diff
@@ -321,99 +317,86 @@ export default class Dao extends Component {
     loadInfos() {
 
         const {
-            addr,
+            addrError,
             noContract,
             noDao,
             noAddProposalRights,
             noApproveProposalRights
         } = this.errors;
+        const { addr, dao } = this.state;
 
-        const multicall = `${this.state.addr.value}.${window.nearConfig.MULTICALL_FACTORY_ADDRESS}`;
-        const dao_address = `${this.state.addr.value}.${SputnikDAO.FACTORY_ADDRESS}`;
-        const dao = new SputnikDAO(dao_address);
+        const multicall = `${addr.value}.${window.nearConfig.MULTICALL_FACTORY_ADDRESS}`;
+        const dao_address = `${addr.value}.${SputnikDAO.FACTORY_ADDRESS}`;
 
-        this.lastAddr = this.state.addr.value;
+        this.lastAddr = addr.value;
 
         noContract.isBad = false;
         noDao.isBad = false;
         noAddProposalRights.isBad = false;
         noApproveProposalRights.isBad = false;
 
-        if (addr.isBad) {
+        // chosen address violates NEAR AccountId rules.
+        if (addrError.isBad) {
             noContract.isBad = true;
             noDao.isBad = true;
             noAddProposalRights.isBad = true;
             noApproveProposalRights.isBad = true;
-            this.setState({ proposed: -1, proposedInfo: {} })
+            this.setState({ proposed: -1, proposedInfo: {} });
+            return;
         }
 
         this.setState({ loading: true });
 
         let newState = {};
 
-        Promise.all([
-            view(multicall, "get_admins", {}).catch(e => {
-                    if (e.type === "AccountDoesNotExist" && e.toString().includes(` ${multicall} `)) {
-                        noContract.isBad = true;
+        // initialize DAO object
+        SputnikDAO.init(dao_address).catch(e => {})
+        .then((newDAO) => {
+            // DAO not ready => either no SputnikDAO contract on the chosen address
+            // or some error happened during DAO object init.
+            this.setState({ dao: newDAO });
+            if (!newDAO.ready) { return; }
```

Comment:
Please print something to the console, so we're not left blind if we happen to break this.

Btw, u can merge this `.then` with the one below, no reason to have them seperate.